tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/ConfigureWrapSpiProvider.java

```diff
@@ -0,0 +1,134 @@
+    private static final String SPI_HEADER = "%s=*".formatted(SPI_PROVIDER);
+
+    @Component
+    private RepositorySystem repoSystem;
+
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true, required = true)
+    private RepositorySystemSession repoSession;
+
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
+    private List<RemoteRepository> repositories;
+
+    @Override
+    protected void onFeatureUpdated(Feature feature) {
+        if (!containsSpiFly(feature)) {
+            // Add spifly as a prerequisite if not exists
+            Dependency dependency = new Dependency("spifly", null);
+            dependency.setPrerequisite(true);
+            feature.getFeature().add(dependency);
+        }
+    }
+
+    private static boolean containsSpiFly(Feature feature) {
+        return feature.getFeature().stream().anyMatch(d -> "spifly".equals(d.getName()));
+    }
+
+    @Override
+    protected boolean processWrappedBundle(Bundle bundle) {
+        String location = bundle.getLocation();
+        Matcher matcher = WRAP_PROTOCOL.matcher(location);
+        if (matcher.matches()) {
+            String groupId = matcher.group(1);
+            String artifactId = matcher.group(2);
+            String version = matcher.group(3);
+            String options = matcher.group(4);
+            if (options != null && options.contains(SPI_PROVIDER)) {
+                return false;
+            } else if (provideSPI(groupId, artifactId, version)) {
+                addHeader(bundle, options, location);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static void addHeader(Bundle bundle, String options, String location) {
+        String separator;
+        if (options == null) {
+            separator = "$";
+        } else if (options.endsWith("&") || options.endsWith("$")) {
+            separator = "";
+        } else {
+            separator = "&";
+        }
+        bundle.setLocation("%s%s%s".formatted(location, separator, SPI_HEADER));
+    }
+
+    private boolean provideSPI(String groupId, String artifactId, String version) {
+        File file = downloadArtifact(groupId, artifactId, version);
+        if (file == null) {
+            getLog().warn("Could not download artifact %s:%s:%s".formatted(groupId, artifactId, version));
+            return false;
+        }
+        try {
+            return containsSPI(file);
+        } catch (IOException e) {
+            getLog().warn("Could not check artifact %s:%s:%s".formatted(groupId, artifactId, version), e);
+        }
+        return false;
+    }
+
+    private static boolean containsSPI(File file) throws IOException {
+        try (ZipFile zip = new ZipFile(file)) {
+            return zip.getEntry("META-INF/services") != null;
+        }
+    }
+
+    private File downloadArtifact(String groupId, String artifactId, String version) {
```

Comment:
maybe rename this method to `resolveArtifact` or `resolveArtifact`?

Reply:
Already fixed thx